xcube/cli/gen.py

```diff
@@ -19,80 +19,86 @@
-                                                 'in append mode, input by input.',
-                                     formatter_class=GenL2CHelpFormatter)
-    parser.add_argument('--version', '-V', action='version', version=version)
-    parser.add_argument('--proc', '-p', dest='input_processor', choices=input_processor_names,
-                        help=f'Input processor type name.')
-    parser.add_argument('--config', '-c', dest='config_file',
-                        help='Data cube configuration file in YAML format.')
-    parser.add_argument('--dir', '-d', dest='output_dir',
-                        help=f'Output directory. Defaults to {DEFAULT_OUTPUT_DIR!r}')
-    parser.add_argument('--name', '-n', dest='output_name',
-                        help=f'Output filename pattern. Defaults to {DEFAULT_OUTPUT_NAME!r}.')
-    parser.add_argument('--writer', '-w', dest='output_writer', choices=output_writer_names,
-                        help=f'Output writer type name. Defaults to {DEFAULT_OUTPUT_WRITER!r}.')
-    parser.add_argument('--size', '-s', dest='output_size',
-                        help='Output size in pixels using format "<width>,<height>".')
-    parser.add_argument('--region', '-r', dest='output_region',
-                        help='Output region using format "<lon-min>,<lat-min>,<lon-max>,<lat-max>"')
-    parser.add_argument('--vars', '-v', dest='output_variables',
-                        help='Variables to be included in output. '
-                             'Comma-separated list of names which may contain wildcard characters "*" and "?".')
-    parser.add_argument('--resamp', dest='output_resampling', choices=resampling_algs,
-                        help='Fallback spatial resampling algorithm to be used for all variables.'
-                             f'Defaults to {DEFAULT_OUTPUT_RESAMPLING!r}.')
-    parser.add_argument('--traceback', dest='traceback_mode', default=False, action='store_true',
-                        help='On error, print Python traceback.')
-    parser.add_argument('--append', '-a', dest='append_mode', default=False, action='store_true',
-                        help='Append successive outputs.')
-    parser.add_argument('--dry-run', default=False, action='store_true',
-                        help='Just read and process inputs, but don\'t produce any outputs.')
-    parser.add_argument('input_files', metavar='INPUT_FILES', nargs='+',
-                        help="One or more input files or a pattern that may contain wildcards '?', '*', and '**'.")
-
-    try:
-        arg_obj = parser.parse_args(args or sys.argv[1:])
-    except SystemExit as e:
-        return int(str(e))
-
-    traceback_mode = arg_obj.traceback_mode
-    append_mode = arg_obj.append_mode
-    dry_run = arg_obj.dry_run
+    traceback_mode = traceback
+    append_mode = append
+    dry_run = dry_run
 
     try:
-        config = get_config_dict(arg_obj, open)
+        config = get_config_dict(locals(), open)
```

Comment:
Why `locals()`?

Reply:
it is now solved by keeping the structure of the old code.